sql/core/src/main/scala/org/apache/spark/sql/execution/SparkOptimizer.scala

```diff
@@ -65,6 +65,8 @@ class SparkOptimizer(
       RewriteDistinctAggregates),
     Batch("Pushdown Filters from PartitionPruning", fixedPoint,
       PushDownPredicates),
+    Batch("Rewrite With expression", fixedPoint,
```

Comment:
shall we just rewrite `With` once at the very end of the optimizer?

Reply:
Let me try.

Reply:
We can't push With filter into scan, so rewrite with at the end of `operatorOptimizationBatch`.